### PR TITLE
Initialize the benchmark tester for operator.

### PR DESCRIPTION
**Pull request by Xreki**
We want to write a common benchmark for all operators so that we can benchmark different operators' performance by providing a config file.

To use this tester to benchmark an operator, you just need to provide a config file, like:
```
$ vim elementwise_add.config
{                                                                                                                                                                                            
  op_type: elementwise_add
  input: { name: X; dims: 64x64 }
  input: { name: Y; dims: 64x1 }
  device_id: 0
  repeat: 100 
  profile: 1
  print_debug_string: 0
}
```

then run,
```
$ your_build_root/paddle/fluid/operators/benchmark/op_tester --filename=path to elementwise_add.config
W0218 08:32:26.152118     9 device_context.cc:263] Please NOTE: device: 0, CUDA Capability: 70, Driver API Version: 9.2, Runtime API Version: 9.2
W0218 08:32:26.152199     9 device_context.cc:271] device: 0, cuDNN Version: 7.0.

------------------------->     Profiling Report     <-------------------------

Place: All
Time unit: ms
Sorted by event first end time in descending order in the same thread

Event                       Calls       Total       Min.        Max.        Ave.        Ratio.      
thread0::elementwise_add    100         2.76202     0.023008    0.114944    0.0276202   1           

I0218 08:32:26.890723     9 op_tester.cc:188] === Run 100 times, latency: 0.0441471 ms ===
```

**TODO (in next PR):**
- [x] Support different data type (float is hard-coded in the current code): **do in #15494**
- [x] Support LoDTensor (lod is not set in the current code) :  **set lod of inputs through config in #15493**
- [x] Support to set `attr`s: **do in #15494**
- [x] Support multiple configs in a single file: **do in #15493**
- [ ] Better output for the statistical purpose
- [ ] more ...

### CODE REVIEW

File: paddle/fluid/operators/benchmark/op_tester.cc [hunk 1]
```diff
@@ -0,0 +1,350 @@
+/* Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserved.
```
Comment (tensor-tang) [inline paddle/fluid/operators/benchmark/op_tester.cc:1]: should be op_benchmark.* ?

File: paddle/fluid/operators/benchmark/op_tester_config.h [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+#if !defined(_WIN32)
+#include <sys/time.h>
+#endif
+#include <chrono>  // NOLINT
+
+namespace paddle {
+namespace operators {
+namespace benchmark {
+
+struct Timer {
```
Comment (tensor-tang) [inline paddle/fluid/operators/benchmark/op_tester_config.h]: Could directly use this ? 
https://github.com/PaddlePaddle/Paddle/blob/6402424f7a82c15ed361d76efcfbb46f79f873a4/paddle/fluid/platform/timer.h#L30

File: paddle/fluid/operators/benchmark/op_tester.cc [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+#include "paddle/fluid/framework/variable_helper.h"
+#include "paddle/fluid/platform/init.h"
+#include "paddle/fluid/platform/profiler.h"
+#include "paddle/fluid/pybind/pybind.h"
+
+namespace paddle {
+namespace operators {
+namespace benchmark {
+
+DEFINE_string(filename, "", "Path of op config file.");
```
Comment (tensor-tang) [inline paddle/fluid/operators/benchmark/op_tester.cc]: `filename` cloud be a name more specific.

File: paddle/fluid/operators/benchmark/op_tester.cc [hunk 4]
```diff
@@ -0,0 +1,350 @@
...
+  // Warm up
+  RunImpl();
+
+  if (config_.profile) {
+    if (platform::is_cpu_place(place_)) {
+      platform::EnableProfiler(platform::ProfilerState::kCPU);
+    } else {
+#ifdef PADDLE_WITH_CUDA
+      platform::EnableProfiler(platform::ProfilerState::kAll);
+      // The default device_id of paddle::platform::CUDAPlace is 0.
```
Comment (tensor-tang) [inline paddle/fluid/operators/benchmark/op_tester.cc]: This comments cloud be options of config, or just remove.

File: paddle/fluid/operators/benchmark/op_tester_config.cc [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+#include "paddle/fluid/operators/benchmark/op_tester_config.h"
+#include <fstream>
+#include "glog/logging.h"
+#include "paddle/fluid/platform/enforce.h"
+
+namespace paddle {
+namespace operators {
+namespace benchmark {
+
+static const char kStartSeparator[] = "{";
```
Comment (tensor-tang) [inline paddle/fluid/operators/benchmark/op_tester_config.cc:24]: What do you mean by `Separator`?

**Review by tensor-tang — COMMENTED** [review]
How to support many op_types in one config file?

File: paddle/fluid/operators/benchmark/op_tester.cc (hunk 3 above)
Comment (Xreki) [reply paddle/fluid/operators/benchmark/op_tester.cc]: Done.

File: paddle/fluid/operators/benchmark/op_tester.cc (hunk 4 above)
Comment (Xreki) [reply paddle/fluid/operators/benchmark/op_tester.cc]: Done.

File: paddle/fluid/operators/benchmark/op_tester_config.cc (hunk 5 above)
Comment (Xreki) [reply paddle/fluid/operators/benchmark/op_tester_config.cc:24]: Just the `separator` for different op configs. A config file may have multiple op configs, but the codes don't support that yet.

File: paddle/fluid/operators/benchmark/op_tester_config.h (hunk 2 above)
Comment (Xreki) [reply paddle/fluid/operators/benchmark/op_tester_config.h]: Done.

File: paddle/fluid/operators/benchmark/op_tester.cc (hunk 1 above)
Comment (Xreki) [reply paddle/fluid/operators/benchmark/op_tester.cc:1]: `op_benchmark` is a little strange...

**Comment by Xreki** [comment]
> How to support many op_types in one config file?

@tensor-tang The design of config support multiple op configs, either the same or different `op_type`s, but the codes don't support yet. May support in next PR. You can provide a config file as:
```
{                                                                                                                                                                                            
  op_type: elementwise_add
  input: { name: X; dims: 64x64 }
  input: { name: Y; dims: 64x1 }
  device_id: 0
  repeat: 100 
  profile: 1
  print_debug_string: 0
}
{
  op_type: mul
  input: ...
  ...
}
```

**Review by tensor-tang — APPROVED** [review]
LGTM

**Pull request merged**